src/rt/cpp/when.h

```diff
@@ -14,13 +15,58 @@ namespace verona::cpp
 {
   using namespace verona::rt;
 
+  template<typename T>
+  struct ActualCownSpan
+  {
+    ActualCown<std::remove_const_t<T>>** array;
+    size_t length;
+
+    void clear()
+    {
+      length = 0;
+      array = nullptr;
+    }
+
+    ActualCownSpan()
+    {
+      clear();
+    }
+
+    ActualCownSpan(ActualCownSpan&& old)
+    {
+      length = old.length;
+      array = old.array;
+      old.clear();
+    }
+
+    ActualCownSpan& operator=(ActualCownSpan&& old)
+    {
+      if (array)
+        snmalloc::ThreadAlloc::get().dealloc(array);

```

Comment:
This is performing deallocation, so doesn't this require ownership?

Reply:
This array holds `ActualCown<T>`, though, not `cown_ptr<T>`. If I understand correctly, when this array is constructed there is no reference count increase.

Reply:
Okay, yeah, so ActualCown is kind of borrowing from the underlying cown_ptr. Makes sense, do you think you could add a comment, so we don't get confused again.  Possibly add a comment to ActualCown too?

Reply:
Should there be a destructor to deallocate the array too?

Reply:
You are right. ActualCown borrows from the underlying cown_array. I'll add the comment.

You are also correct about the confusion with the dealloc. This is wrong and it shouldn't be there. The lifetime of the ActualCownSpan is always the same as the AccessBatch. The memory the span uses is managed by the AccessBatch constructors and destructors. There is no need for a move assignment operator. I'll remove that.

The lifetime of the Access and AccessBatch goes all the way till after the execution of the behaviour.

Reply:
Given that this the case, should the ActualCownSpan be defined inside the AccessBatch?

Reply:
Yeah.  So `ActualCownSpan` is different to `ActualCown`.  `ActualCown` is the Verona runtimes view of a cown, where as `ActualCownSpan` is part of the behaviour structure. I think moving it into AccessBatch makes sense, I also wonder if it should be `Actual...`, but differently named?

Reply:
Actually, it should be `ActualCownPtrSpan` since it's an array of pointers to `ActualCown`.